unison-core/src/Unison/NameSegment.hs

```diff
@@ -27,6 +27,14 @@ segments' n = go split
     go ("" : z) = go z
     go (x : y) = x : go y
 
+
+-- | Splits a textual name into name segments, returns a boolean indicating whether the
+-- name was absolute.
+splitText :: Text -> (Bool, [NameSegment])
```

Comment:
Note, this doesn't handle the name `.`; but that will be [removed from base](https://github.com/unisonweb/base/issues/50) very soon. I plan to also make it an invalid name 😄